contracts/libraries/Position.sol

```diff
@@ -31,7 +31,16 @@ library Position {
         view
         returns (Position.Info storage position)
     {
-        position = self[keccak256(abi.encodePacked(owner, tickLower, tickUpper))];
+        // positionKey = keccak256(abi.encodePacked(owner, tickLower, tickUpper))
+        bytes32 positionKey;
+        /// @solidity memory-safe-assembly
+        assembly {
+            mstore(0x06, tickUpper) // [0x23, 0x26)
+            mstore(0x03, tickLower) // [0x20, 0x23)
+            mstore(0, owner) // [0x0c, 0x20)
```

Comment:
is it OK to just start writing to memory address 0 though? this is an internal function and will take on the context of wherever it's called from

Reply:
The scratch space is specifically used for abi encoding and `keccak256`, like computing the storage slot for a key in a mapping, according to the [docs](https://docs.soliditylang.org/en/latest/internals/layout_in_memory.html). Any correctly written EVM code shouldn't have assumptions about values in the scratch space.